### ISSUE TITLE
`lisp-repl` is broken on master

### ISSUE THREAD

**Issue by aadcg** [atlas-engineer/nyxt#2369]
Try to evaluate any s-exp on the REPL.

```
<WARN> [21:39:18] Warning: JavaScript error: GError: Domain: "WebKitJavascriptError", Code: 699, Message: nyxt:nyxt/repl-mode:lisp-repl:1:60: TypeError: null is not an object (evaluating 'document.querySelector('.input-buffer[data-repl-id=\"0\"]').focus')
<WARN> [21:39:18] Disabling other repl instance is already in buffer #<WEB-BUFFER 3004 {100D5B36A3}>
; in: LAMBDA ()
;     (SPINNERET:WITH-HTML-STRING
;       (:HEAD (:STYLE (NYXT:STYLE NYXT/REPL-MODE:REPL-MODE)))
;       (:BODY
;        (:DIV :ID "container"
;         (:DIV :ID "evaluations"
;          (LOOP NYXT/REPL-MODE::FOR NYXT/REPL-MODE:EVALUATION NYXT/REPL-MODE::IN #
;                NYXT/REPL-MODE::FOR NYXT/REPL-MODE::ORDER NYXT/REPL-MODE::FROM 0
;                NYXT/REPL-MODE::COLLECT #
;                NYXT/REPL-MODE::COLLECT ...)
;          (:DIV :CLASS "input" # #)))))
; --> SB-INT:DX-LET LET SB-INT:DX-LET LET LET SPINNERET:WITH-HTML LET PROGN 
; --> SPINNERET::WITH-TAG PROG1 LET PROGN FLET #:<BODY>22 BLOCK 
; --> SPINNERET::CATCH-OUTPUT SPINNERET:HTML LOCALLY LET SPINNERET::WITH-TAG 
; --> PROG1 LET PROGN FLET #:|<DIV#container>29| BLOCK SPINNERET::CATCH-OUTPUT 
; --> SPINNERET:HTML SPINNERET::WITH-TAG PROG1 LET PROGN FLET 
; --> #:|<DIV#evaluations>35| BLOCK SPINNERET::CATCH-OUTPUT SPINNERET:HTML LOOP 
; --> BLOCK LET LET SB-LOOP::WITH-LOOP-LIST-COLLECTION-HEAD LET* TAGBODY 
; --> SB-LOOP::LOOP-COLLECT-RPLACD RPLACD 
; ==>
;   1
; 
; caught WARNING:
;   VALUE is not a valid attribute for <TEXTAREA>
; 
; caught WARNING:
;   VALUE is not a valid attribute for <TEXTAREA>
; 
; caught WARNING:
;   VALUE is not a valid attribute for <TEXTAREA>
; 
; compilation unit finished
;   caught 3 WARNING conditions
```

**Comment by aadcg** [comment]
After a (long) git bissect session here's what I got:

```
There are only 'skip'ped commits left to test.
The first bad commit could be any of:
5cf78e7ff9962e805ee14b5939154c34636b06c2
dde6643791b29e1eccc55ee1acf37810eb76ad17
cfa612513acc223da899ed6f951e397d0aa007cf
e5d329132942f036e6cbbd59b42bed86e1e951d0
7b51d23fc7bd9583f745aaa414b4f57353affc4e
7b95a0c8e049465ae0c858bf836e5aeada64cbaf
a9f625c4c45e88328365c42d6acd89125aa04b9c
ac6e76bf4d0f0338d294c6cd2feb6abfd71f0eb5
2447c863a0695ddb7e06fbc54b1c221baf054915
987a960f2a4a87f07c4dd3d0d142d51bc86ff41b
1220b7de8be7529f51d3bacb750d11000b7e5f25
ca0f2edb0774af311a1190d0fd66b44eb123e088
325e7d826a2eff09c10b378905d09f457d5c0ce2
fc0727a615b904c02ada0687d4b6f78932ff13cf
4b2f0282001df453def561d901c5fe72e670d450
617ad79be46481e4d8422f92be4ef6dd9aa0e6d0
7f78cbe25176787ddf532500375c19c8c0367f20
bf7daf01fea4f95ab751d10a8f2bc5156127e737
84804894abfd4b349fa0729a850cc0eff173fed6
e87ad4017b3ab7c1e76517bfb216fc2e7b19c4bf
432ca62268c1ab98965d16cff23fff5428976413
f0f74cba910d9958d6a42f17f9a702419c560253
18cf52fe1fcd43e7f4263881c1bf776bf4cd194e
e42ad03ea67077d3462241fb24a3167c6d660e56
0c74ef8bb7f2aef9b5d71b368dc0d43bd2c76d2f
c1843f18d5f8b236cc06ddc2d69f6c77d0728b47
599013b7a32c69ad4105e3933fec011587b4ffd9
ac81be9b2f8165936cf647124794e2c8c748a88f
4ad92256d50156ec3ebd833f8728274e474c8a79
6c2ebe8c3a5f6836d845d62f5e5afefb5a3218cb
1e1801e97402ffe5fff3137471637015feaf025a
229cb51c5b1a574bab953e0bf6a0167478876ebc
1138feae291c55b3fd7b063e01129b7e6a441bb4
2271adf78feefd7e9f5d428820c0648754fabfee
83bfe84ee4488a4d8465f66ab4fde54e147f8f8b
efa7f204591ea65c15fe756c649652037544c51b
7714377b17f3097881b8a1ccd5ddf8ffb148d464
8fc7ce0f52396b7fbe7c25836d288b8cc8584082
5746eea3e101317779aa558b4aa361328d535700
5d307d0b365f08c574b3504d309afc2fb2a3ec72
3e930dfc9de877e6f9e44ffb5f0cffd88717eae8
962311d067816de653d1f88ce10f3a02f0b8b552
0e189d8a0271b5f214ed54de87862981daeac001
ccf461af5c114ccbee2381b06151fc8cd6c79096
80351b3bfe20978d20447c1eba3e79610f6f840b
We cannot bisect more!
```

```
Bisect Log:
git bisect start 'master' 'ccc9e97d2'
4ed6f26ee bad mode/document: Replace ID argument with BUFFER in summarize-buffer.
ccc9e97d2 good mode/repl: Use HTML <textarea> instead of <input> to allow for multilines.

git bisect skip 7f78cbe25176787ddf532500375c19c8c0367f20
7f78cbe25 skip Fix lisp-eval2 load order vs. lisp-url-callbacks.

git bisect skip 7714377b17f3097881b8a1ccd5ddf8ffb148d464
7714377b1 skip GI-GTK: Ignore unused arguments.

git bisect good 07ede9a240596915ecca45d1b94a31536353d82c
07ede9a24 good Call print-status from setf :after methods.

git bisect skip 325e7d826a2eff09c10b378905d09f457d5c0ce2
325e7d826 skip renderer-script: Remove `form' writer duplicate.

git bisect skip 5cf78e7ff9962e805ee14b5939154c34636b06c2
5cf78e7ff skip urls: Fix lisp-eval2 macro.

git bisect skip 83bfe84ee4488a4d8465f66ab4fde54e147f8f8b
83bfe84ee skip inspector: Move "Change value" to the right of the symbol, not of that value.

git bisect skip dde6643791b29e1eccc55ee1acf37810eb76ad17
dde664379 skip renderer-script: Fix anonymous internal-page visibility.

git bisect good 4ea63b2bea1658bb8f06eb60a9ef4c17947d17ab
4ea63b2be good nyxt.asd: Detail where to we install.

git bisect skip 7b51d23fc7bd9583f745aaa414b4f57353affc4e
7b51d23fc skip renderer-script: Define reinitialize-instance for internal-page.

git bisect skip ac81be9b2f8165936cf647124794e2c8c748a88f
ac81be9b2 skip buffer: Replace ID argument with BUFFER in switch-buffer and delete-buffer.

git bisect skip 2447c863a0695ddb7e06fbc54b1c221baf054915
2447c863a skip command: Explain why we use `defgeneric'.

git bisect skip 2271adf78feefd7e9f5d428820c0648754fabfee
2271adf78 skip renderer-script: Rename `mode' slot to `page-mode'.

git bisect skip ac6e76bf4d0f0338d294c6cd2feb6abfd71f0eb5
ac6e76bf4 skip renderer-script: Fix internal-page form definition.

git bisect skip 229cb51c5b1a574bab953e0bf6a0167478876ebc
229cb51c5 skip mode: Fix enable-modes and disable-modes to not block REPL.

git bisect skip 1220b7de8be7529f51d3bacb750d11000b7e5f25
1220b7de8 skip Define reduce-to-buffer in time.

git bisect skip 18cf52fe1fcd43e7f4263881c1bf776bf4cd194e
18cf52fe1 skip Document show-url-qrcode and define-internal-page.

git bisect skip cfa612513acc223da899ed6f951e397d0aa007cf
cfa612513 skip renderer-script: Fix argument passing.

git bisect skip 4ad92256d50156ec3ebd833f8728274e474c8a79
4ad92256d skip parenscript-macro: Remove unused funcall-ps-callback.

git bisect skip e5d329132942f036e6cbbd59b42bed86e1e951d0
e5d329132 skip urls: Fix lisp2 title fetching.

git bisect skip 432ca62268c1ab98965d16cff23fff5428976413
432ca6226 skip inspector: Use interned symbols instead of uninterned symbols.

git bisect skip 7b95a0c8e049465ae0c858bf836e5aeada64cbaf
7b95a0c8e skip buffer: Fix reload-buffers when called with singleton.

git bisect skip 599013b7a32c69ad4105e3933fec011587b4ffd9
599013b7a skip command: Fix (make-instance 'command ...) to return functional command.

git bisect bad 37a61ccca3c6bd0eb4e952212a747093bad48ee8
37a61ccca bad mode/history: Replace ID argument with BUFFER in buffer-history-tree.

git bisect skip bf7daf01fea4f95ab751d10a8f2bc5156127e737
bf7daf01f skip Rename lisp-eval2 to lisp-eval.

git bisect good 1df274325a9d2b2be4ef569de219d739bcb16749
1df274325 good makefile: Have `install' depend on `all'.

git bisect skip c1843f18d5f8b236cc06ddc2d69f6c77d0728b47
c1843f18d skip renderer-script: have internal-page inherit from `command'.

git bisect good 4f115db65668b2e0fd761593a87c804ec300768e
4f115db65 good foreign-interface: Add text to clipboard-ring after copying/pasting.

git bisect good 30140ab09c73ce79cbf158fb415963b672de948e
30140ab09 good Define new lisp:// URL which support full Lisp bodies, including closures.

git bisect skip ccf461af5c114ccbee2381b06151fc8cd6c79096
ccf461af5 skip GTK: Allow for live updates of the scheme.

git bisect skip 617ad79be46481e4d8422f92be4ef6dd9aa0e6d0
617ad79be skip describe: Rename describe-function argument `function' to `fn'.

git bisect skip 0e189d8a0271b5f214ed54de87862981daeac001
0e189d8a0 skip Decouple define-internal-page from define-internal-page-command.

git bisect skip 4b2f0282001df453def561d901c5fe72e670d450
4b2f02820 skip parenscript-macro: Remove obsolete `lisp-eval'.

git bisect skip 3e930dfc9de877e6f9e44ffb5f0cffd88717eae8
3e930dfc9 skip Define command manually for internal pages using non-scalar parameters.

git bisect skip fc0727a615b904c02ada0687d4b6f78932ff13cf
fc0727a61 skip mode/buffer-listing: Fix panel-buffers reference in `buffers' panel.

git bisect skip 5d307d0b365f08c574b3504d309afc2fb2a3ec72
5d307d0b3 skip urls: Fix page-name package in parse-nyxt-url.

git bisect skip ca0f2edb0774af311a1190d0fd66b44eb123e088
ca0f2edb0 skip nyxt.asd: Load inspector afer nyxt/ps:lisp-eval2.

git bisect skip 5746eea3e101317779aa558b4aa361328d535700
5746eea3e skip Merge branch 'lisp-nyxt-urls-refactor2'

git bisect skip 987a960f2a4a87f07c4dd3d0d142d51bc86ff41b
987a960f2 skip urls: Fix lisp-eval2 default buffer.

git bisect skip 8fc7ce0f52396b7fbe7c25836d288b8cc8584082
8fc7ce0f5 skip renderer-script: Fix argument evaluation in define-internal-page.

git bisect skip 84804894abfd4b349fa0729a850cc0eff173fed6
84804894a skip Fix lisp-eval in status-buffer.

git bisect skip efa7f204591ea65c15fe756c649652037544c51b
efa7f2045 skip Move *nyxt-url-commands* to renderer-script to be defined in time.

git bisect skip a9f625c4c45e88328365c42d6acd89125aa04b9c
a9f625c4c skip Turn all lisp-eval to lisp-eval2.

git bisect skip 1138feae291c55b3fd7b063e01129b7e6a441bb4
1138feae2 skip inspector: Fix value->html :fn argument.

git bisect skip e87ad4017b3ab7c1e76517bfb216fc2e7b19c4bf
e87ad4017 skip describe: Change lisp-eval2 argument to `buffer' instead of callback-index.

git bisect skip 962311d067816de653d1f88ce10f3a02f0b8b552
962311d06 skip browser: Switch to numerical IDs.

git bisect skip f0f74cba910d9958d6a42f17f9a702419c560253
f0f74cba9 skip describe: Fix macro expansion of function argument.

git bisect skip 1e1801e97402ffe5fff3137471637015feaf025a
1e1801e97 skip describe: Remove describe-value command, keep internal-page only.

git bisect skip e42ad03ea67077d3462241fb24a3167c6d660e56
e42ad03ea skip mode/document: Use new syntax for define-internal-page.

git bisect good df6cfd50e6050c55de54475c15c4a7cb9b8fbe4c
df6cfd50e good GTK: Call buffer-load instead of ffi-buffer-load.

git bisect skip 6c2ebe8c3a5f6836d845d62f5e5afefb5a3218cb
6c2ebe8c3 skip manual: Don't use JavaScript when dumped.

git bisect skip 0c74ef8bb7f2aef9b5d71b368dc0d43bd2c76d2f
0c74ef8bb skip mode/os-package-manager: Decouple describe-os-package internal-page from command.

git bisect bad 80351b3bfe20978d20447c1eba3e79610f6f840b
80351b3bf bad urls: Fix page-mode enabling in "nyxt" URLs.

git bisect bad 80351b3bfe20978d20447c1eba3e79610f6f840b
# only skipped commits left to test
# possible first bad commit: [80351b3bfe20978d20447c1eba3e79610f6f840b] urls: Fix page-mode enabling in "nyxt" URLs.
# possible first bad commit: [ccf461af5c114ccbee2381b06151fc8cd6c79096] GTK: Allow for live updates of the scheme.
# possible first bad commit: [962311d067816de653d1f88ce10f3a02f0b8b552] browser: Switch to numerical IDs.
# possible first bad commit: [5746eea3e101317779aa558b4aa361328d535700] Merge branch 'lisp-nyxt-urls-refactor2'
# possible first bad commit: [8fc7ce0f52396b7fbe7c25836d288b8cc8584082] renderer-script: Fix argument evaluation in define-internal-page.
# possible first bad commit: [83bfe84ee4488a4d8465f66ab4fde54e147f8f8b] inspector: Move "Change value" to the right of the symbol, not of that value.
# possible first bad commit: [1138feae291c55b3fd7b063e01129b7e6a441bb4] inspector: Fix value->html :fn argument.
# possible first bad commit: [1e1801e97402ffe5fff3137471637015feaf025a] describe: Remove describe-value command, keep internal-page only.
# possible first bad commit: [6c2ebe8c3a5f6836d845d62f5e5afefb5a3218cb] manual: Don't use JavaScript when dumped.
# possible first bad commit: [ac81be9b2f8165936cf647124794e2c8c748a88f] buffer: Replace ID argument with BUFFER in switch-buffer and delete-buffer.
# possible first bad commit: [0c74ef8bb7f2aef9b5d71b368dc0d43bd2c76d2f] mode/os-package-manager: Decouple describe-os-package internal-page from command.
# possible first bad commit: [18cf52fe1fcd43e7f4263881c1bf776bf4cd194e] Document show-url-qrcode and define-internal-page.
# possible first bad commit: [432ca62268c1ab98965d16cff23fff5428976413] inspector: Use interned symbols instead of uninterned symbols.
# possible first bad commit: [84804894abfd4b349fa0729a850cc0eff173fed6] Fix lisp-eval in status-buffer.
# possible first bad commit: [bf7daf01fea4f95ab751d10a8f2bc5156127e737] Rename lisp-eval2 to lisp-eval.
# possible first bad commit: [4b2f0282001df453def561d901c5fe72e670d450] parenscript-macro: Remove obsolete `lisp-eval'.
# possible first bad commit: [fc0727a615b904c02ada0687d4b6f78932ff13cf] mode/buffer-listing: Fix panel-buffers reference in `buffers' panel.
# possible first bad commit: [ca0f2edb0774af311a1190d0fd66b44eb123e088] nyxt.asd: Load inspector afer nyxt/ps:lisp-eval2.
# possible first bad commit: [987a960f2a4a87f07c4dd3d0d142d51bc86ff41b] urls: Fix lisp-eval2 default buffer.
# possible first bad commit: [a9f625c4c45e88328365c42d6acd89125aa04b9c] Turn all lisp-eval to lisp-eval2.
# possible first bad commit: [7b95a0c8e049465ae0c858bf836e5aeada64cbaf] buffer: Fix reload-buffers when called with singleton.
# possible first bad commit: [e5d329132942f036e6cbbd59b42bed86e1e951d0] urls: Fix lisp2 title fetching.
# possible first bad commit: [5cf78e7ff9962e805ee14b5939154c34636b06c2] urls: Fix lisp-eval2 macro.
# possible first bad commit: [dde6643791b29e1eccc55ee1acf37810eb76ad17] renderer-script: Fix anonymous internal-page visibility.
# possible first bad commit: [cfa612513acc223da899ed6f951e397d0aa007cf] renderer-script: Fix argument passing.
# possible first bad commit: [7b51d23fc7bd9583f745aaa414b4f57353affc4e] renderer-script: Define reinitialize-instance for internal-page.
# possible first bad commit: [ac6e76bf4d0f0338d294c6cd2feb6abfd71f0eb5] renderer-script: Fix internal-page form definition.
# possible first bad commit: [2447c863a0695ddb7e06fbc54b1c221baf054915] command: Explain why we use `defgeneric'.
# possible first bad commit: [1220b7de8be7529f51d3bacb750d11000b7e5f25] Define reduce-to-buffer in time.
# possible first bad commit: [325e7d826a2eff09c10b378905d09f457d5c0ce2] renderer-script: Remove `form' writer duplicate.
# possible first bad commit: [617ad79be46481e4d8422f92be4ef6dd9aa0e6d0] describe: Rename describe-function argument `function' to `fn'.
# possible first bad commit: [7f78cbe25176787ddf532500375c19c8c0367f20] Fix lisp-eval2 load order vs. lisp-url-callbacks.
# possible first bad commit: [e87ad4017b3ab7c1e76517bfb216fc2e7b19c4bf] describe: Change lisp-eval2 argument to `buffer' instead of callback-index.
# possible first bad commit: [f0f74cba910d9958d6a42f17f9a702419c560253] describe: Fix macro expansion of function argument.
# possible first bad commit: [e42ad03ea67077d3462241fb24a3167c6d660e56] mode/document: Use new syntax for define-internal-page.
# possible first bad commit: [c1843f18d5f8b236cc06ddc2d69f6c77d0728b47] renderer-script: have internal-page inherit from `command'.
# possible first bad commit: [599013b7a32c69ad4105e3933fec011587b4ffd9] command: Fix (make-instance 'command ...) to return functional command.
# possible first bad commit: [4ad92256d50156ec3ebd833f8728274e474c8a79] parenscript-macro: Remove unused funcall-ps-callback.
# possible first bad commit: [229cb51c5b1a574bab953e0bf6a0167478876ebc] mode: Fix enable-modes and disable-modes to not block REPL.
# possible first bad commit: [2271adf78feefd7e9f5d428820c0648754fabfee] renderer-script: Rename `mode' slot to `page-mode'.
# possible first bad commit: [efa7f204591ea65c15fe756c649652037544c51b] Move *nyxt-url-commands* to renderer-script to be defined in time.
# possible first bad commit: [7714377b17f3097881b8a1ccd5ddf8ffb148d464] GI-GTK: Ignore unused arguments.
# possible first bad commit: [5d307d0b365f08c574b3504d309afc2fb2a3ec72] urls: Fix page-name package in parse-nyxt-url.
# possible first bad commit: [3e930dfc9de877e6f9e44ffb5f0cffd88717eae8] Define command manually for internal pages using non-scalar parameters.
# possible first bad commit: [0e189d8a0271b5f214ed54de87862981daeac001] Decouple define-internal-page from define-internal-page-command.
# bad: [80351b3bfe20978d20447c1eba3e79610f6f840b] urls: Fix page-mode enabling in "nyxt" URLs.
# only skipped commits left to test
# possible first bad commit: [80351b3bfe20978d20447c1eba3e79610f6f840b] urls: Fix page-mode enabling in "nyxt" URLs.
# possible first bad commit: [ccf461af5c114ccbee2381b06151fc8cd6c79096] GTK: Allow for live updates of the scheme.
# possible first bad commit: [962311d067816de653d1f88ce10f3a02f0b8b552] browser: Switch to numerical IDs.
# possible first bad commit: [5746eea3e101317779aa558b4aa361328d535700] Merge branch 'lisp-nyxt-urls-refactor2'
# possible first bad commit: [8fc7ce0f52396b7fbe7c25836d288b8cc8584082] renderer-script: Fix argument evaluation in define-internal-page.
# possible first bad commit: [83bfe84ee4488a4d8465f66ab4fde54e147f8f8b] inspector: Move "Change value" to the right of the symbol, not of that value.
# possible first bad commit: [1138feae291c55b3fd7b063e01129b7e6a441bb4] inspector: Fix value->html :fn argument.
# possible first bad commit: [1e1801e97402ffe5fff3137471637015feaf025a] describe: Remove describe-value command, keep internal-page only.
# possible first bad commit: [6c2ebe8c3a5f6836d845d62f5e5afefb5a3218cb] manual: Don't use JavaScript when dumped.
# possible first bad commit: [ac81be9b2f8165936cf647124794e2c8c748a88f] buffer: Replace ID argument with BUFFER in switch-buffer and delete-buffer.
# possible first bad commit: [0c74ef8bb7f2aef9b5d71b368dc0d43bd2c76d2f] mode/os-package-manager: Decouple describe-os-package internal-page from command.
# possible first bad commit: [18cf52fe1fcd43e7f4263881c1bf776bf4cd194e] Document show-url-qrcode and define-internal-page.
# possible first bad commit: [432ca62268c1ab98965d16cff23fff5428976413] inspector: Use interned symbols instead of uninterned symbols.
# possible first bad commit: [84804894abfd4b349fa0729a850cc0eff173fed6] Fix lisp-eval in status-buffer.
# possible first bad commit: [bf7daf01fea4f95ab751d10a8f2bc5156127e737] Rename lisp-eval2 to lisp-eval.
# possible first bad commit: [4b2f0282001df453def561d901c5fe72e670d450] parenscript-macro: Remove obsolete `lisp-eval'.
# possible first bad commit: [fc0727a615b904c02ada0687d4b6f78932ff13cf] mode/buffer-listing: Fix panel-buffers reference in `buffers' panel.
# possible first bad commit: [ca0f2edb0774af311a1190d0fd66b44eb123e088] nyxt.asd: Load inspector afer nyxt/ps:lisp-eval2.
# possible first bad commit: [987a960f2a4a87f07c4dd3d0d142d51bc86ff41b] urls: Fix lisp-eval2 default buffer.
# possible first bad commit: [a9f625c4c45e88328365c42d6acd89125aa04b9c] Turn all lisp-eval to lisp-eval2.
# possible first bad commit: [7b95a0c8e049465ae0c858bf836e5aeada64cbaf] buffer: Fix reload-buffers when called with singleton.
# possible first bad commit: [e5d329132942f036e6cbbd59b42bed86e1e951d0] urls: Fix lisp2 title fetching.
# possible first bad commit: [5cf78e7ff9962e805ee14b5939154c34636b06c2] urls: Fix lisp-eval2 macro.
# possible first bad commit: [dde6643791b29e1eccc55ee1acf37810eb76ad17] renderer-script: Fix anonymous internal-page visibility.
# possible first bad commit: [cfa612513acc223da899ed6f951e397d0aa007cf] renderer-script: Fix argument passing.
# possible first bad commit: [7b51d23fc7bd9583f745aaa414b4f57353affc4e] renderer-script: Define reinitialize-instance for internal-page.
# possible first bad commit: [ac6e76bf4d0f0338d294c6cd2feb6abfd71f0eb5] renderer-script: Fix internal-page form definition.
# possible first bad commit: [2447c863a0695ddb7e06fbc54b1c221baf054915] command: Explain why we use `defgeneric'.
# possible first bad commit: [1220b7de8be7529f51d3bacb750d11000b7e5f25] Define reduce-to-buffer in time.
# possible first bad commit: [325e7d826a2eff09c10b378905d09f457d5c0ce2] renderer-script: Remove `form' writer duplicate.
# possible first bad commit: [617ad79be46481e4d8422f92be4ef6dd9aa0e6d0] describe: Rename describe-function argument `function' to `fn'.
# possible first bad commit: [7f78cbe25176787ddf532500375c19c8c0367f20] Fix lisp-eval2 load order vs. lisp-url-callbacks.
# possible first bad commit: [e87ad4017b3ab7c1e76517bfb216fc2e7b19c4bf] describe: Change lisp-eval2 argument to `buffer' instead of callback-index.
# possible first bad commit: [f0f74cba910d9958d6a42f17f9a702419c560253] describe: Fix macro expansion of function argument.
# possible first bad commit: [e42ad03ea67077d3462241fb24a3167c6d660e56] mode/document: Use new syntax for define-internal-page.
# possible first bad commit: [c1843f18d5f8b236cc06ddc2d69f6c77d0728b47] renderer-script: have internal-page inherit from `command'.
# possible first bad commit: [599013b7a32c69ad4105e3933fec011587b4ffd9] command: Fix (make-instance 'command ...) to return functional command.
# possible first bad commit: [4ad92256d50156ec3ebd833f8728274e474c8a79] parenscript-macro: Remove unused funcall-ps-callback.
# possible first bad commit: [229cb51c5b1a574bab953e0bf6a0167478876ebc] mode: Fix enable-modes and disable-modes to not block REPL.
# possible first bad commit: [2271adf78feefd7e9f5d428820c0648754fabfee] renderer-script: Rename `mode' slot to `page-mode'.
# possible first bad commit: [efa7f204591ea65c15fe756c649652037544c51b] Move *nyxt-url-commands* to renderer-script to be defined in time.
# possible first bad commit: [7714377b17f3097881b8a1ccd5ddf8ffb148d464] GI-GTK: Ignore unused arguments.
# possible first bad commit: [5d307d0b365f08c574b3504d309afc2fb2a3ec72] urls: Fix page-name package in parse-nyxt-url.
# possible first bad commit: [3e930dfc9de877e6f9e44ffb5f0cffd88717eae8] Define command manually for internal pages using non-scalar parameters.
# possible first bad commit: [0e189d8a0271b5f214ed54de87862981daeac001] Decouple define-internal-page from define-internal-page-command.
```

**Comment by aadcg** [comment]
I call on all of my respectable hackers to pause for a moment and reflect on the consequences of the way we're dealing with our git log. @Ambrevar @aartaka @jmercouris

**Comment by Ambrevar** [comment]
I don't understand.  The bug was introduced with 5746eea3e101317779aa558b4aa361328d535700, the merge of lisp-nyxt-url-refactor2.  It's very recent and a single commit.

What was the problem?

**Comment by Ambrevar** [comment]
Bug is fixed with 6a66eff95e0ad54f17e2920adcf652ebab3114a1.

Regarding your Git issue, I still do not understand what you mean or what you were trying to do.  If you think there is an issue with the workflow, feel free to open a new issue about it :)

**Comment by jmercouris** [comment]
@aadcg do you think we should not rebase? Or do you think we should maintain branches? Is it something different than we discussed via email? We are always eager to improve!
